content/en/docs/concepts/storage/volumes.md

```diff
@@ -1391,72 +963,489 @@ plugin path on each node and in some cases the control plane nodes as well.
+
+#### Creating a VMDK volume  {#creating-vmdk-volume}
+
+Choose one of the following methods to create a VMDK.
+
+{{< tabs name="tabs_volumes" >}}
+{{% tab name="Create using vmkfstools" %}}
+First ssh into ESX, then use the following command to create a VMDK:
+
+```shell
+vmkfstools -c 2G /vmfs/volumes/DatastoreName/volumes/myDisk.vmdk
+```
+{{% /tab %}}
+{{% tab name="Create using vmware-vdiskmanager" %}}
+Use the following command to create a VMDK:
+
+```shell
+vmware-vdiskmanager -c -t 0 -s 40GB -a lsilogic myDisk.vmdk
+```
+
+{{% /tab %}}
+
+{{< /tabs >}}
+
+
+#### vSphere VMDK configuration example {#vsphere-vmdk-configuration}
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-vmdk
+spec:
+  containers:
+  - image: k8s.gcr.io/test-webserver
+    name: test-container
+    volumeMounts:
+    - mountPath: /test-vmdk
+      name: test-volume
+  volumes:
+  - name: test-volume
+    # This VMDK volume must already exist.
+    vsphereVolume:
+      volumePath: "[DatastoreName] volumes/myDisk"
+      fsType: ext4
+```
+
+For more information, see the [vSphere volume]](https://github.com/kubernetes/examples/tree/master/staging/volumes/vsphere) examples.
+
+#### vSphere CSI migration {#vsphere-csi-migration}
+
+{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+
+The `CSIMigration` feature for vsphereVolume, when enabled, redirects all plugin operations
+from the existing in-tree plugin to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver. In order to use this feature, the [vSphere CSI
+Driver](https://github.com/kubernetes-sigs/vsphere-csi-driver)
+must be installed on the cluster and the `CSIMigration` and `CSIMigrationvSphere`
+[feature gates](/docs/reference/command-line-tools-reference/feature-gates/) must be enabled.
+
+This also requires minimum vSphere vCenter/ESXi Version to be 7.0u1 and minimum HW Version to be VM version 15.
+
+{{< note >}}
+The following StorageClass parameters from the built-in `vsphereVolume` plugin are not supported by the vSphere CSI driver:
+
+* `diskformat`
+* `hostfailurestotolerate`
+* `forceprovisioning`
+* `cachereservation`
+* `diskstripes`
+* `objectspacereservation`
+* `iopslimit`
+
+Existing volumes created using these parameters will be migrated to the vSphere CSI driver, but new volumes created by the vSphere CSI driver will not be honoring these parameters.
+{{< /note >}}
+
+#### vSphere CSI migration complete {#vsphere-csi-migration-complete}
+{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+
+To turn off the `vsphereVolume` plugin from being loaded by the controller manager and the kubelet, you need to set this feature flag to `true`. You must install a `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver on all worker nodes.
 
-```shell
-sudo systemctl daemon-reload
-sudo systemctl restart docker
-```
 
 ## {{% heading "whatsnext" %}}
 
```

Comment:
I would direct the reader to read https://k8s.io/docs/concepts/storage/persistent-volumes/